app/scripts/modules/core/src/search/externalSearch.registry.ts

```diff
@@ -20,7 +20,7 @@ export interface IExternalSearchConfig {
    * Method to fetch search results
    * @param query
    */
-  search: (query: string) => IPromise<ISearchResult[]>;
+  search: (query: string | { [key: string]: string }) => IPromise<ISearchResult[]>;
```

Comment:
Since the new search API can take a map of key/value filter pairs, we need to update the search API in deck to handle this and pass along the filters accordingly.